executor/src/witgen/generator.rs

```diff
@@ -59,15 +61,18 @@ impl<'a, T: FieldElement> Machine<'a, T> for Generator<'a, T> {
             .cloned()
             .unwrap_or_else(|| self.compute_partial_first_row(mutable_state));
 
-        let ProcessResult { eval_value, block } =
-            self.process(first_row, 0, mutable_state, Some(outer_query), false);
+        let ProcessResult {
+            eval_value,
+            updated_data,
+        } = self.process(first_row, 0, mutable_state, Some(outer_query), false);
 
         let eval_value = if eval_value.is_complete() {
             log::trace!("End processing VM '{}' (successfully)", self.name());
             // Remove the last row of the previous block, as it is the first row of the current
             // block.
             self.data.pop();
-            self.data.extend(block);
+            self.data.extend(updated_data.block);
+            self.publics.extend(updated_data.publics);
```

Comment:
This is so horrible for performance...
At some point we need to make a big refactor and allow direct access to all those variables from within the processors (unless they are really isolated).

Reply:
Ah ok, if it's only run at the end then it's fine. I was not worried about the size of the publics but that we have yet another check for every regular cell value we set.